---
title: A tarefa DataStore GarbageCollection falha com erro
description: Descrição
solution: Experience Manager
product: Experience Manager
applies-to: Experience Manager
keywords: KCS
resolution: Resolution
internal-notes: null
bug: false
article-created-by: Sunali Paul
article-created-date: 8/3/2022 10:43:31 AM
article-published-by: Sunali Paul
article-published-date: 8/5/2022 7:27:13 AM
version-number: 1
article-number: KA-20258
dynamics-url: https://adobe-ent.crm.dynamics.com/main.aspx?forceUCI=1&pagetype=entityrecord&etn=knowledgearticle&id=9174741c-1913-ed11-b83d-0022480867fb
exl-id: 841377b0-1f7c-4026-9de9-838bf88c0aa0
source-git-commit: e8f4ca2dd578944d4fe399074fab461de88ad247
workflow-type: tm+mt
source-wordcount: '316'
ht-degree: 0%

---

# A tarefa DataStore GarbageCollection falha com erro

## Descrição


DataStoreGarbageCollectionTask falha com o erro abaixo

\*ERRO\* sling-oak-observation-19 org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector Nem todos os repositórios marcaram referências disponíveis : e0fb2a15-d5d5-4013-9607-0fb89cf87c94 26.07.2022 20:56:50.737 \*ERRO\* sling-oak-observation-19 org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector Erro de coleta de lixo Blob java.io.IOException: Nem todos os repositórios marcaram referências disponíveis em org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector$GarbageCollectionType$1.mergeAllMarkedReferences(MarkSweepGarbageCollector.java:7 22) org.apache.jackrabbit.oak-blob-plugins:1.8 21 em org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector.sweep(MarkSweepGarbageCollector.java:384) org.apache.jackrabbit.oak-blob-plugins :1.8.21 em org.apache.jackrabbit.oak.plugins.org blob.MarkSweepGarbageCollector.markAndSweep(MarkSweepGarbageCollector.java:284) org.apache.jackrabbit.oak-blob-plugins:1.8.21 em org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector.collect Garbage(MarkSweepGarbageCollector.java:191) org.apache.jackrabbit.oak-blo b-plugins:1.8.21 em org.apache.jackrabbit.oak.plugins.blob.BlobGC$1.call(BlobGC.java:87) org.apache.jackrabbit oak-blob-plugins:1.8.21 em org.apache.jackrabbit.oak.plugins.blob.BlobGC$1.call(BlobGC.java:83) org.apache.jackrabbit.oak-blob-plugins:1.8.21 em java.util.concurrent.FutureTask.run FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java .lang.Thread.run(Thread.java:748) 26.07.2022 20:56:51.620 \*ERRO\* pool-20-thread-1 org.apache.jackrabbit.oak.commons.jmx.ManagementOperation Falha na coleta de lixo do Blob java.io.IOException: Nem todos os repositórios marcaram referências disponíveis em org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector$GarbageCollectionType$1.mergeAllMarkedReferences(MarkSweepGarbageCollector.java:7 22) org.apache.jackrabbit.oak-blob-plugins:1.8 21 em org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector.sweep(MarkSweepGarbageCollector.java:384) org.apache.jackrabbit.oak-blob-plugins :1.8.21 em org.apache.jackrabbit.oak.plugins.org blob.MarkSweepGarbageCollector.markAndSweep(MarkSweepGarbageCollector.java:284) org.apache.jackrabbit.oak-blob-plugins:1.8.21 em org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector.collect Garbage(MarkSweepGarbageCollector.java:191) org.apache.jackrabbit.oak-blo b-plugins:1.8.21 em org.apache.jackrabbit.oak.plugins.blob.BlobGC$1.call(BlobGC.java:87) org.apache.jackrabbit oak-blob-plugins:1.8.21 em org.apache.jackrabbit.oak.plugins.blob.BlobGC$1.call(BlobGC.java:83) org.apache.jackrabbit.oak-blob-plugins:1.8.21 em java.util.concurrent.FutureTask.run FutureTask.java:266) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) at java .lang.Thread.run(Thread.java:748) 26.07.2022 20:56:51.628 \*INFO\* pool-20-thread-1 com.adobe.granite.maintenance.impl.MaintenanceJobsManagerImpl Name=&#39;DataStoreGarbageCollectionTask&#39;, Status =&#39;ERROR&#39;, Result=&#39;Data store GC: Falha na operação da coleta de lixo do Blob: Nem todos os repositórios marcaram referências disponíveis&#39;, Details=&#39;{&quot;created&quot; :1658858205603,&quot;started&quot;:1658858205603, &quot;completed&quot;:1658858216 20,&quot;duration&quot;:6017}&#39;


## Resolução


Esse problema pode ocorrer se você tiver uma instância anterior usando o armazenamento de dados, que foi destruído e uma nova instância criada para usar o armazenamento de dados.

O problema pode ser resolvido após a remoção da ID do repositório registrada no armazenamento de dados. O GC do armazenamento de blobs deve ser concluído com êxito.

Guia passo a passo: 1) Vá para o caminho do armazenamento de dados aem-install-folder/crx-quickstart/repository/datastore e pesquise a ID do repositório <b>e0fb2a15-d5d5-4013-9607-0fb89cf87c94</b> (deve ser igual ao mencionado no error.log <b>ERRO</b>\* sling-oak-observation-19 org.apache.jackrabbit.oak.plugins.blob.MarkSweepGarbageCollector Nem todos os repositórios marcaram referências disponíveis : <b>e0fb2a15-d5d5-4013-9607-0fb89cf87c94</b>), você terá um arquivo de referência com esse nome, basta excluir esse arquivo de referência (repita essas etapas se houver vários arquivos).

2) Execute o GC novamente e verifique se ele resolve o problema.
